frame-reader.go

```diff
@@ -0,0 +1,150 @@
+// Read a generic "framed" packet consisting of a header and a
+// This is used for both TLS Records and TLS Handshake Messages
+package mint
+
+import (
+	"fmt"
+	"io"
+)
+
+type frameDetails interface {
+	headerLen() int
+	defaultReadLen() int
+	frameLen(hdr []byte) (int, error)
+}
+
+const (
+	kFrameReaderHdr  = 0
+	kFrameReaderBody = 1
+)
+
+var frameReaderWouldBlock = fmt.Errorf("Would have blocked")
+
+type frameNextAction func(f *frameReader) error
+
+type frameReader struct {
+	conn        io.Reader
```

Comment:
I did some of this.

I don't think you want to try to find *all* the frames because returning a bunch of frames is hard to manage, and you still need to know whether more data is needed to complete some partial frame. So, I did the first but not the second.